src/Stan/Observation.hs

```diff
@@ -26,8 +34,68 @@ data Observation = Observation
     , observationInspectionId :: !(Id Inspection)
     , observationLoc          :: !RealSrcSpan
     , observationFile         :: !FilePath
+    , observationModuleName   :: !Text
+    , observationFileContent  :: !ByteString
     } deriving stock (Show, Eq)
 
 -- | Show 'Observation' in a human-friendly format.
 prettyShowObservation :: Observation -> Text
-prettyShowObservation = show
+prettyShowObservation Observation{..} = unlines $
+    map (" ┃  " <>)
+        $  observationTable
+        <> ("" : source)
+        <> ("" : solution)
+  where
+    observationTable :: [Text]
+    observationTable =
+        [ element "ID:            " <> b (unId observationId)
+        , element "Severity:      " <> prettyShowSeverity (inspectionSeverity inspection)
+        , element "Description:   " <> inspectionDescription inspection
+        , element "Inspection ID: " <> unId observationInspectionId
+        , element "Category:      " <> categories
+        ]
+      where
+        element :: Text -> Text
+        element = formatWith [italic] . ("✦ " <>)
+
+        b :: Text -> Text
+        b = formatWith [bold]
+
+    inspection :: Inspection
+    inspection = getInspectionById observationInspectionId
+
+    categories :: Text
+    categories = Text.intercalate " "
+        $ map prettyShowCategory $ NE.toList $ inspectionCategory inspection
+
+    source :: [Text]
+    source =
+        [ alignLine (n - 1)
+        , alignLine n <> getSourceLine
+        , alignLine (n + 1) <> arrows
+        ]
+      where
+        n :: Int
+        n = srcSpanStartLine observationLoc
+
+        alignLine :: Int -> Text
+        alignLine x = Text.justifyRight 4 ' ' (show x) <> " ┃ "
+
+        getSourceLine :: Text
+        getSourceLine = decodeUtf8 $
+            BS.lines observationFileContent !! (n - 1)
```

Comment:
I agree that the usage of partial functions here is okay. It's too much work to handle errors in here and report them, especially when they shouldn't happen..